src/mongocxx/database.hpp

```diff
@@ -248,6 +248,7 @@ class MONGOCXX_API database {
 
     class MONGOCXX_PRIVATE impl;
     std::unique_ptr<impl> _impl;
+    const std::unique_ptr<impl>& get_impl() const;
```

Comment:
I think get_impl() is fine, hooray for moving away from ambiguous names!

Will callers of this method ever want to alter the impl's state?  It seems ok to let them do so, in which case maybe it's better to return an impl* than a const unique_ptr&